search-provider/eks-search-provider.c

```diff
@@ -196,6 +197,18 @@ do_search (EksSearchProvider *self,
            GDBusMethodInvocation *invocation,
            gchar **terms)
 {
+  EkncEngine *engine = eknc_engine_get_default ();
```

Comment:
Is this necessary?

I don't think we ever had problems before with apps not showing up in desktop search if they had never been run.

Reply:
Its necessary in the sense that all the other eknservices interfaces are broken without something like this. I can also imagine why SDK1 apps would be broken without it too - not sure how those symlinks would even be created?

Reply:
I just tested with an SDK1 app, and this is indeed not necessary. Something (although I'm not sure what) is already symlinking the shards when you do a desktop search. I'd say let's remove this, so that we do as few operations as possible when starting up the search providers.